cobalt/watchdog/watchdog.h

```diff
@@ -139,8 +159,11 @@ class Watchdog : public Singleton<Watchdog> {
   SbTimeMonotonic time_last_written_microseconds_ = 0;
   // Number of microseconds between writes.
   int64_t write_wait_time_microseconds_;
-  // Dictionary of registered Watchdog clients.
+  // Dictionary of name registered Watchdog clients.
   std::unordered_map<std::string, std::unique_ptr<Client>> client_map_;
+  // List of client registered Watchdog clients, parallel data structure to
+  // client_map_.
+  std::vector<std::shared_ptr<Client>> client_list_;
```

Comment:
I get a little bit confused. Do we want to keep both client_map_ and client_list_? Are they independent to each other? I feel the new data structure is enough to support all the use cases, do we have a plan to remove client_map_ in future?

Reply:
Right now we are keeping both client_map_ and client_list_. The original implementation uses unique key names where as the new implementation uses client pointers so it completely lacks unique identifiers that base::Value can support. Thus a simple client_list_ is required for the new implementation. The 2 data structures can not be easily combined without something horribly complicated. I think it's both simpler and much cleaner that the 2 data structures remain separate.

Reply:
Yeah, I mean do we want to deprecate the original implementation and only use the new client_list_ in future?

Reply:
no because the original implementation still relies on a dictionary and not a list. it still has unique named clients it expects can be used as a key. it cannot iterate over a list where there may be multiple clients with the same name.